### ISSUE TITLE
Extract Info

### ISSUE THREAD

**Issue by mady143** [OmkarPathak/ResumeParser#34]
Hi @OmkarPathak  @dazzag24 ,
I had clone your data and started parsing the resume but i found some difficulty in extracting the experience and company names can you tell how it works from utils.py file from the functions extract_experience and extract_competencies 

Thanks and Regards,
Manikantha Sekhar

**Comment by OmkarPathak** [comment]
@mady143 yes it is very difficult to parse experience and company names. The model that we have trained is not yet accurate in extracting these entities. If you want, you can train your model and then try the same

**Comment by mady143** [comment]
@OmkarPathak  okay how to train the model is their any training process in existing code 
and 
also i was unable  to print the data from the some other functions which are in different files only getting the prints from views.py can you explain me y?

Thanks and Regards,
Manikantha Sekhar

**Comment by OmkarPathak** [comment]
> okay how to train the model

There are these two files: [train](https://github.com/OmkarPathak/pyresparser/blob/master/pyresparser/custom_train.py) and [test](https://github.com/OmkarPathak/pyresparser/blob/master/pyresparser/custom_t.py).
Your training data should be in the format like [this](https://github.com/OmkarPathak/pyresparser/blob/master/pyresparser/traindata.json)

> also i was unable to print the data from the some other functions which are in different files only getting the prints from views.py can you explain me y?

can you elaborate this issue?

**Comment by mady143** [comment]
Hi @OmkarPathak ,
I had done training process like ,
**(my_resume_pyresparser) manikantha@Manikantha-PC:~/Downloads/pyresparser/pyresparser$ python custom_train.py** 
Created blank 'en' model
Creating new pipe
Starting iteration 0
Losses {'ner': 83948.83401187272}
Starting iteration 1
Losses {'ner': 50266.19444030836}
Starting iteration 2
Losses {'ner': 43155.15530648954}
Starting iteration 3
Losses {'ner': 40022.855776749966}
Starting iteration 4
Losses {'ner': 34466.00632723602}
Starting iteration 5
Losses {'ner': 34260.82662641417}
Starting iteration 6
Losses {'ner': 32415.686345038288}
Starting iteration 7
Losses {'ner': 30070.207913930943}
Starting iteration 8
Losses {'ner': 28438.445806700598}
Starting iteration 9
Losses {'ner': 28697.96626680234}
Starting iteration 10
Losses {'ner': 24849.52654336362}
Starting iteration 11
Losses {'ner': 25142.6057196642}
Starting iteration 12
Losses {'ner': 24079.14422368379}
Starting iteration 13
Losses {'ner': 24257.180968177847}
Starting iteration 14
Losses {'ner': 23090.409918362162}
Starting iteration 15
Losses {'ner': 23996.485883846104}
Starting iteration 16
Losses {'ner': 23423.123708186868}
Starting iteration 17
Losses {'ner': 20781.21235375348}
Starting iteration 18
Losses {'ner': 20025.49499945144}
Starting iteration 19
Losses {'ner': 19859.159115096158}
Starting iteration 20
Losses {'ner': 19835.303925276472}
Starting iteration 21
Losses {'ner': 17874.102371228983}
Starting iteration 22
Losses {'ner': 18699.1655548761}
Starting iteration 23
Losses {'ner': 18402.408249338278}
Starting iteration 24
Losses {'ner': 17600.381271163213}
Starting iteration 25
Losses {'ner': 17745.068060417587}
Starting iteration 26
Losses {'ner': 17847.67826112547}
Starting iteration 27
Losses {'ner': 16500.602022483115}
Starting iteration 28
Losses {'ner': 17123.114352572746}
Starting iteration 29
Losses {'ner': 17659.68112649223}
Entities in 'Marathwada Mitra Mandals College of Engineering'
Name Marathwada Mitra Mandals College of Engineering
Saved model to /home/manikantha/Downloads/pyresparser/pyresparser
Loading from /home/manikantha/Downloads/pyresparser/pyresparser
Name Marathwada Mitra Mandals College of Engineering
and later i run custom_t.py 
as 
**(my_resume_pyresparser) manikantha@Manikantha-PC:~/Downloads/pyresparser/pyresparser$ python custom_t.py**
i am getting **{}** empty dict
so can you tell me how to resolve the issue...

Thanks and Regards,
Manikantha Sekhar..

**Comment by mady143** [comment]
Hi @OmkarPathak 
actually i used pyresparser to extract the Resume and get the data from that though i am getting the all required data like 
name,
email,
mobile_number,
education(degree)
designation,
and skills
but in this also missing the experience so how can i get the experience can you explain me 

Thanks and Regrads,
Manikantha Sekhar...

**Comment by mady143** [comment]
> Hi @OmkarPathak ,
> I had done training process like ,
> **(my_resume_pyresparser) manikantha@Manikantha-PC:~/Downloads/pyresparser/pyresparser$ python custom_train.py**
> Created blank 'en' model
> Creating new pipe
> Starting iteration 0
> Losses {'ner': 83948.83401187272}
> Starting iteration 1
> Losses {'ner': 50266.19444030836}
> Starting iteration 2
> Losses {'ner': 43155.15530648954}
> Starting iteration 3
> Losses {'ner': 40022.855776749966}
> Starting iteration 4
> Losses {'ner': 34466.00632723602}
> Starting iteration 5
> Losses {'ner': 34260.82662641417}
> Starting iteration 6
> Losses {'ner': 32415.686345038288}
> Starting iteration 7
> Losses {'ner': 30070.207913930943}
> Starting iteration 8
> Losses {'ner': 28438.445806700598}
> Starting iteration 9
> Losses {'ner': 28697.96626680234}
> Starting iteration 10
> Losses {'ner': 24849.52654336362}
> Starting iteration 11
> Losses {'ner': 25142.6057196642}
> Starting iteration 12
> Losses {'ner': 24079.14422368379}
> Starting iteration 13
> Losses {'ner': 24257.180968177847}
> Starting iteration 14
> Losses {'ner': 23090.409918362162}
> Starting iteration 15
> Losses {'ner': 23996.485883846104}
> Starting iteration 16
> Losses {'ner': 23423.123708186868}
> Starting iteration 17
> Losses {'ner': 20781.21235375348}
> Starting iteration 18
> Losses {'ner': 20025.49499945144}
> Starting iteration 19
> Losses {'ner': 19859.159115096158}
> Starting iteration 20
> Losses {'ner': 19835.303925276472}
> Starting iteration 21
> Losses {'ner': 17874.102371228983}
> Starting iteration 22
> Losses {'ner': 18699.1655548761}
> Starting iteration 23
> Losses {'ner': 18402.408249338278}
> Starting iteration 24
> Losses {'ner': 17600.381271163213}
> Starting iteration 25
> Losses {'ner': 17745.068060417587}
> Starting iteration 26
> Losses {'ner': 17847.67826112547}
> Starting iteration 27
> Losses {'ner': 16500.602022483115}
> Starting iteration 28
> Losses {'ner': 17123.114352572746}
> Starting iteration 29
> Losses {'ner': 17659.68112649223}
> Entities in 'Marathwada Mitra Mandals College of Engineering'
> Name Marathwada Mitra Mandals College of Engineering
> Saved model to /home/manikantha/Downloads/pyresparser/pyresparser
> Loading from /home/manikantha/Downloads/pyresparser/pyresparser
> Name Marathwada Mitra Mandals College of Engineering
> and later i run custom_t.py
> as
> **(my_resume_pyresparser) manikantha@Manikantha-PC:~/Downloads/pyresparser/pyresparser$ python custom_t.py**
> i am getting **{}** empty dict
> so can you tell me how to resolve the issue...
> 
> Thanks and Regards,
> Manikantha Sekhar..

I resolved this issue by increasing the iterations so at first i tried for 30 then empty dict as an output and later i was tried for 50 then i got the output a little bit so their is an issue to resolve is increase the iterations

**Comment by mady143** [comment]
Hi @OmkarPathak ,
How can i link the train and test to this folder to get the accurate experience of the candidate in the resume and display in the browser

![screen1001](https://user-images.githubusercontent.com/37180858/76841447-c89a6e80-685e-11ea-937a-2effd1b2dc3e.png)

in the above image i have to display the experience also before training and testing i was unable to get the experience so i remove the experience column 
but now i am running the training and testing files from another github code Pyreparser (https://github.com/OmkarPathak/pyresparser) so i how could i put these files in https://github.com/OmkarPathak/ResumeParser/ in this project 

please help me in this 

Thanks & Regards,
Manikantha Sekhar..

**Comment by OmkarPathak** [comment]
@mady143 you can do it using `python setup.py install`. This will install pyresparser using your pre-trained model

**Comment by mady143** [comment]
@OmkarPathak ,
https://github.com/OmkarPathak/pyresparser how can i test these code through UI to get the result as like your previous code in https://github.com/OmkarPathak/ResumeParser/ to get the experience and company names....

**Comment by OmkarPathak** [comment]
@mady143 for that you need the knowledge of django. You can add you're fields in [views.py](https://github.com/OmkarPathak/ResumeParser/blob/master/resume_parser/parser_app/views.py) and then render them in [template](https://github.com/OmkarPathak/ResumeParser/blob/master/resume_parser/parser_app/templates/base.html)

**Comment by mady143** [comment]
Hi @OmkarPathak ,
Actually i am developing a module to match the job description we have to get the matched resume and also matches the experience though for that i had used pyresparser package for that in some cases or some resume i am getting the total_experience as 4.7 like that but missing the experience but giving the content with month year its unable to calculate the overall years 
can you explain in a simple way either with training process or as in normal way  can you help me in this or you can given any related  links which matches for my requirement...

Thanks and Regards,
Manikantha Sekhar...
